src/EnergyPlus/General.cc

```diff
@@ -333,10 +513,23 @@ namespace General {
 			DY = Y0 - Y1;
 			if ( std::abs( DY ) < SMALL ) DY = SMALL;
 			// new estimation
-			XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			if ( ( HVACSystemRootFinding.TypeNum == Bisection ) ||
+				( HVACSystemRootFinding.TypeNum == Alternation && AltIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == RegulaFalsiThenBisection && NIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == BisectionThenRegulaFalsi && NIte <= HVACSystemRootFinding.NumOfIter ) ) {
+				if ( HVACSystemRootFinding.TypeNum == Alternation ) {
+					if ( AltIte >= 2 * HVACSystemRootFinding.NumOfIter ) AltIte = 0;
+				}
+				// Bisection
+				XTemp = ( X1 + X0 ) / 2.0;
+			} else {
+				// Regula Falsi
+				XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			}
```

Comment:
OK. I will make the change. Before I do it. Here is what Rich suggested:

        switch ( HVACSystemRootFinding.TypeNum )

        {
            case RegulaFalsi : {
                XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
            }
            case Bisection : {
                XTemp = ( X1 + X0 ) / 2.0;
                break;
            }
            case Alternation : {
                if ( AltIte > HVACSystemRootFinding.NumOfIter ) {
                   if ( AltIte >= 2 * HVACSystemRootFinding.NumOfIter ) AltIte = 0;
                   XTemp = ( X1 + X0 ) / 2.0;
                } else {
                   XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
                }
            }
            case ...
        }

@jasondegraw Do I have to use enum class for Switch statements?